qiskit/algorithms/time_evolvers/variational/solvers/ode/ode_function_factory.py

```diff
@@ -0,0 +1,82 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Abstract class for generating ODE functions."""
+from __future__ import annotations
+
+from abc import ABC
+from enum import Enum
+from typing import Callable
+
+import numpy as np
+
+from qiskit.circuit import Parameter
+
+from .abstract_ode_function import AbstractOdeFunction
+from .ode_function import OdeFunction
+
+from ..var_qte_linear_solver import VarQTELinearSolver
+
+
+class OdeFunctionType(Enum):
+    """Types of ODE functions for VatQTE algorithms."""
+
+    # more will be supported in the near future
```

Comment:
```suggestion
    # Other types may be supported in the future
```
My preference I think would be to remove the comment and not make any implicit commitment to supporting other types. If you feel that something is needed then maybe more toned down as per the suggestion.